.buildkite/scripts/package.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+##
+##  Builds the distribution packages: tarballs, RPMs, Docker images, etc.
+##  It runs for both linux/amd64 and linux/arm64. On linux/amd64 it builds all packages;
+##  on linux/arm64 it builds only Docker images.
+##
+
+set -eo pipefail
+
+TYPE="$1"
+PLATFORM_TYPE=$(uname -m)
+
+MAKE_GOAL=package
+if [[ ${PLATFORM_TYPE} == "arm" || ${PLATFORM_TYPE} == "aarch64" ]]; then
+  MAKE_GOAL=package-docker
+fi
+
+if [[ ${TYPE} == "snapshot" ]]; then
+  MAKE_GOAL="${MAKE_GOAL}-snapshot"
+fi
+
+echo "--- Configure golang :golang:"
+eval "$(./gvm $(cat .go-version))"
```

Comment:
Ideally the PATH should updated but for some reason `.buildkite/hooks/pre-command` does not export env variables accordingly.

For instance `GO_VERSION` is exported , see https://github.com/elastic/apm-server/pull/11400/files#diff-71e4f7cc0d357183cbbb900bb97a29db006263fb1967d06035e417c38f27e53dR80, but it cannot be used with something like

```suggestion
eval "$(./gvm $GO_VERSION)"
```